js/client-mainmenu.js

```diff
@@ -114,7 +132,140 @@
+				}
+			}
+			});
+			observer.observe(document.activeElement, config);
+			},
+		onUserLogIn:function(self){
+			var buf = '<div class="challenge"><form class="battleform"><p>Challenge from ' + BattleLog.escapeHTML(window.Storage.prefs("challenger")) + '?</p>';
+			buf += '<p><label class="label">Format:</label>' + self.renderFormats(window.Storage.prefs("format")) + '</p>';
+			buf += '<p id="select-team"><label class="label">Team:</label>' + self.renderTeams(window.Storage.prefs("format")) + '</p>';
+			buf +='<button type="button" id=accept-challenge>accept</button>';
+			buf +='<button type="button" id=close-challenge>close</button></form></div>'
+			$('.pmbox').append("<div class=challenge-container>"+buf+"</div>");
+			$('.select.formatselect').prop("disabled",true);
+			$('.select.teamselect').prop("disabled",false);
+			$('#accept-challenge').click(function(){
+				self.onTeamSelect(window.Storage.prefs("challenger"),self);
+			})
+			$('#close-challenge').click(function(){
+				$('.challenge-container').empty();
+			})
+		},
+		onTeamSelect:function(challengerName,self){
+			$('#accept-challenge').html("processing request...");
+				if(window.Storage.prefs("team")!="random"){
+					window.Storage.prefs("validateTeam","customteam")
+					app.sendTeam(window.Storage.teams[$('.select.teamselect').val()]);
+					app.send('/vtm ' + $('.select.formatselect').val());
+				}else{
+					$('.challenge-container').empty();
+					app.send('/utm null');
+					self.sendChallenge({userId:challengerName,format:$('.select.formatselect').val()});
+				}
+		},
+		validateChallengeAndSend:function(self,data){
+			$('#accept-challenge').html("accept");
+					if(data.massage.includes("Your team is valid") && data.userId!=window.app.user.get("name") && data.userId){
+						window.Storage.prefs("validateTeam",null);
+						self.sendChallenge({userId:data.userId,format:$('.select.formatselect').val()});
+						$('.challenge-container').empty();
+					 }else if(data.userId==app.user.get("name")){
+						window.Storage.prefs("validateTeam",null);
+						window.app.addPopupMessage("opponent name and challeger name  are equall you cant challenge your self");
+					 }else{
+						window.Storage.prefs("validateTeam",null);
+						window.app.addPopupMessage(data.massage);
+					}
+		},
+		Popuplesiner:function(self,msg){
+		   //when validating team you ether recive rejection popup or validation pop up messege 
+			if(msg.includes("popup")){
+				window.Storage.prefs("errormassege",msg);
+				//validate team after receving validation messege from server when sending team
+				if(window.Storage.prefs("validateTeam")=="customteam"){
+					self.validateChallengeAndSend(self,{userId:window.Storage.prefs("challenger"),massage:msg});
+				}
+			}
+		},
+		generateLink:function(challengeData){
+			if(!$('.link').length){
+			let link="https://"+window.Config.routes.client+"?challenge-"+challengeData.userId+"-"+challengeData.format+"-"+challengeData.teamIndex;
```

Comment:
Why declare this twice? You never change it, so why not just declare it above the if statement?